src/cli/commands/why.js

```diff
@@ -255,9 +288,14 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
     }
   };
 
-  reporter.step(4, 4, reporter.lang('whyCalculating'), emoji.get('aerial_tramway'));
+  // Only show fourth step if the size flag is passed
+  if (flags.withSize) {
+    steps.push({msg: reporter.lang('whyCalculating'), emoji: emoji.get('aerial_tramway'), action: () => {}});
```

Comment:
You need to do `steps.push()` for this before your first `displayStep()` call for it to show the right number of steps on the console.